src/libDirectoryService/DSBlockPreProcessing.cpp

```diff
@@ -666,6 +675,18 @@ bool DirectoryService::DSBlockValidator(
         return false;
     }
 
+    if (m_mediator.m_curSWInfo != m_pendingDSBlock->GetHeader().GetSWInfo())
+    {
+        auto func = [this]() mutable -> void {
```

Comment:
any race condition analysis of the memory access in this lambda function?

Reply:
OK, I will add some protection on m_mediator.m_curSWInfo